### DISCUSSION TITLE
Resetting Inputs is broken for <p:commandButton/> resetValues="true" Flag. - PrimeFaces 14.0.0

### DISCUSSION THREAD

**Discussion by hangrybear666** [primefaces/community#2200] [PrimeFaces] [answered]
In my entire Application, `resetValues="true"` Flags on `<p:commandButton/>` Elements have no effect since upgrading from PrimeFaces 10. 
I have to manually call `PrimeFaces.current().resetInputs(resetInputId);` in my models and pass the IDs I want to update to the model as well to achieve the desired result.

The `resetValues` Flag in the following code has no effect:
```
<p:commandButton global="false" class="generic-btn-cancel"
        value="#{commonRes.common_cancel}" icon="pi pi-undo"
        actionListener="#{exampleModel.cancelEditOrInsert}"
        update="resetInputId"
        immediate="true" resetValues="true"/>
```

@tandraschko I've created a primefaces-test project that is self-explanatory.

**MP4 Screen Capture displaying workflow:**
https://github.com/primefaces/community/assets/85507965/a8045e9c-3993-4a89-9fab-da53c5fb9ab4

**Reproducer Primfaces v14 Test Archive**
[primefaces-test-commandbutton-resetValues.zip](https://github.com/user-attachments/files/16071343/primefaces-test-commandbutton-resetValues.zip)

P.S. ignore the src\main\webapp\effectiveDataTable.xhtml file, it's for another test project I'm making

**Comment by tandraschko** [comment]
we would need a reproducer primefaces-test project

**Reply by hangrybear666** [reply]
> we would need a reproducer primefaces-test project

Can do next week. Just now entered vacation after migrating from v10 to v14 for two weeks straight. Cheers

**Reply by melloware** [reply]
Created a Spec Issue: https://github.com/jakartaee/faces/issues/1936

**Reply by hangrybear666** [reply]
Thanks. I feel honored my bug report is reaching BalusC.

FWIW:
When I refactor to `<p:resetInput target="frmTest"/>` instead of using the `resetValues` Option the code across my app works perfectly again.

**Reply by melloware** [reply]
Yep because that uses the PF custom code that recurses the tree where the native JSF code appears not to be.

**Comment by vieira-brz** [comment]
It looks like you're encountering an issue with the `resetValues` attribute of the `<p:commandButton>` component in PrimeFaces after upgrading from version 10. If `resetValues="true"` no longer has the desired effect, there could be several reasons for this behavior, such as changes in the framework's handling of component state resets or perhaps some conflicts with other attributes.

Here’s a more informal but structured approach to troubleshooting and potentially resolving this issue:

### 1. **Check for Breaking Changes and Compatibility**
Start by looking through the PrimeFaces release notes between your previous version (10) and your current version. It's crucial to identify any breaking changes related to the `resetValues` attribute or the lifecycle of command buttons. Sometimes, changes in a framework’s behavior are documented, and the solutions are mentioned in these notes.

### 2. **Review Component Usage**
Verify that the `resetValues` attribute is being used correctly:
- Ensure that `immediate="true"` is necessary in your context because setting `immediate` to `true` can alter the phase during which values are processed, potentially affecting when resets occur.
- Check if the `update` attribute points to the correct components that need resetting. Sometimes, naming or scoping changes might cause updates to miss their targets.

### 3. **Test Without `immediate="true"`**
Try removing `immediate="true"` temporarily to see if it affects the behavior of `resetValues`. Immediate can sometimes interfere with the usual JSF lifecycle in a way that might prevent some updates or resets from functioning as expected.

```xml
<p:commandButton global="false" class="generic-btn-cancel"
        value="#{commonRes.common_cancel}" icon="pi pi-undo"
        actionListener="#{exampleModel.cancelEditOrInsert}"
        update="resetInputId"
        resetValues="true"/>
```

### 4. **Debugging and Manual Resets**
If the automated reset still doesn't work, implementing a manual reset via backing bean methods as you've described is a reliable workaround. Make sure your manual reset method covers all cases where resets are necessary. It’s a bit more code but ensures control over the process.

### 5. **Community and Support**
If this still doesn’t resolve the issue, consider reaching out to the PrimeFaces community or checking Stack Overflow for similar issues. Often, other developers might have encountered the same issue and found a solution or workaround.

### 6. **Fallback Plan**
As a last resort, if the issue significantly impacts your application and cannot be resolved in a timely manner, consider the feasibility of either rolling back to a previous version of PrimeFaces where this was not an issue or waiting for a patch if this is identified as a bug by the developers.

### Conclusion
It's annoying when something that worked just fine suddenly doesn't after an upgrade! While the `resetValues` attribute should work as documented, changes in newer versions of frameworks can sometimes lead to unexpected behavior. The manual reset method you’re using is solid for a quick fix, but keep an eye out for any updates or patches from PrimeFaces that might address this issue in a more integrated manner.

**Comment by tandraschko** [comment]
Generated answer?

**Comment by hangrybear666** [comment]
@tandraschko I've created a primefaces-test project that is self-explanatory.

**MP4 Screen Capture displaying workflow:**
https://github.com/primefaces/community/assets/85507965/a8045e9c-3993-4a89-9fab-da53c5fb9ab4

**Reproducer Primfaces v14 Test Archive**
[primefaces-test-commandbutton-resetValues.zip](https://github.com/user-attachments/files/16071343/primefaces-test-commandbutton-resetValues.zip)

P.S. ignore the src\main\webapp\effectiveDataTable.xhtml file, it's for another test project I'm making

**Reply by melloware** [reply, marked as answer]
@hangrybear666 i converted this to a PF issue here: https://github.com/primefaces/primefaces/issues/12241  I would subscribe to that issue.

**Reply by hangrybear666** [reply]
Thank you for your help. Appreciate it.
